test/python/circuit/test_singleton.py

```diff
@@ -325,20 +326,42 @@ def __init__(self, x):
         self.assertEqual(gate.x, 1)
         self.assertIsNot(MyAbstractGate(1), MyAbstractGate(1))
 
-    def test_inherit_singleton(self):
-        class Measure(SingletonInstruction):
-            def __init__(self):
-                super().__init__("measure", 1, 1, [])
+    def test_return_type_singleton_instructions(self):
+        measure = Measure()
+        self.assertEqual(measure.__class__.__name__, "_SingletonMeasure")
+
+        reset = Reset()
+        self.assertEqual(reset.__class__.__name__, "_SingletonReset")
 
+    def test_singleton_instruction_integration(self):
+        qc = QuantumCircuit(1, 1)
+        measure = Measure()
+        reset = Reset()
+        qc.append(measure, [0], [0])
+        qc.append(reset, [0])
+        self.assertEqual(qc.data[0].operation.__class__.__name__, "_SingletonMeasure")
+        self.assertEqual(qc.data[1].operation.__class__.__name__, "_SingletonReset")
```

Comment:
For this test, the intention was to test the `QuantumCircuit.measure` and `QuantumCircuit.reset` methods, rather than the direct use of `Measure` and `Reset` constructors, which we tested in the test above this one.